packages/app/features/home/searchBar.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useState } from 'react';
+import { View, TextInput, Button } from 'react-native';
+import styles from './styles';
+
+const SearchBar = ({ onButtonPress }) => {
```

Comment:
Make your components Typesafe, otherwise who's using it could pass anything to this `onButtonPress` prop

Reply:
In that case, the type should be const? Or () => {}?

Reply:
`const` or `function` has nothing to do with the argument and return type. In other words, you could type this correctly with both approaches. I prefere using the `const` with `lambda`

Here's a way to do it for React Components.
```tsx
interface SearchBarProps {
    onButtonPress: () => void;
}

const SearchBar: React.FC<SearchBarProps> = ({ onButtonPress }) => {
    //...
}
```

The `React.FC` is a way to type the props and also the return ( which is a JSX Element )

Reply:
done